sdk/metrics-testing/src/main/java/io/opentelemetry/sdk/testing/assertj/metrics/AbstractPointDataAssert.java

```diff
@@ -47,4 +49,24 @@ public AttributesAssert attributes() {
     isNotNull();
     return OpenTelemetryAssertions.assertThat(actual.getAttributes());
   }
+
+  /** Returns convenience API to assert against the {@code exemplars} field. */
+  public AbstractIterableAssert<?, ? extends Iterable<? extends Exemplar>, Exemplar, ?>
```

Comment:
This goes back to my comment on the other PR, at the time it seemed to still be ok having only one, but now we have `attributes` and `exemplars` which can't both be called on the same point. This makes me strongly advocate for `hasExemplarsSatisfying` type of methods instead, or otherwise delete them for now instead of locking them into the API with the next release.

Reply:
I had forgotten the metrics-testing artifact is still alpha, so there is no urgent need to delete methods

Reply:
TL;DR; I'd rather not detract from the important bits of the CL for cosmetic testing concerns, so if you really want to remove it, that's fine, but i"d ask you to wait for the new metrics SDK implementation to land. 

That said, here's my opinion:

The convention of having convenient mechanisms of altering the "Assertion" class is core to Assertj design right now, and we have to deal with.  E.g. AbstractIterable supports an "extracting" method that flips the thing you're asserting on in an incompatible way.  I've made use of this in some scenarios and it's highly convenient.   Even if you don't allow these methods in OTel-provided APIs, they still exist (and are expected) from upstream Assertj.  Do you plan to prevent their usage as well?

If this is just a matter of my choosing bad names, I'm used to dealing with my poor naming and am happy to fix it.  However the notion that you shouldn't flip the underlying assertion class.... Well yes it's not ideal but that ship has sailed in choosing Assertj.  In fact, sometimes it's really important convenient.  Specifically, you probably want to go take a second look here: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics-testing/src/main/java/io/opentelemetry/sdk/testing/assertj/metrics/MetricDataAssert.java#L99

When it comes to readability, feel free to look at usages like this: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/DoubleCounterSdkTest.java#L73

Again, happy to remove all these methods if you think they'll lead to major issues in practice, but I don't really think that they hurt readability or correctness.   They *do* have a wart if you want to look at two collections in the same assert, but then you can just go to the more verbose syntax.